### DISCUSSION TITLE
Concept Resources

### DISCUSSION THREAD

**Discussion by expede** [robot-overlord/witchcraft#115] [General]
A few people have asked for resources on the concepts that Witchcraft is porting. This thread is an ongoing collection of these.

* Functors, Applicatives, And Monads In Pictures [link 4](http://adit.io/posts/2013-04-17-functors,_applicatives,_and_monads_in_pictures.html)
* Haskell from First Principles [homepage 3](https://haskellbook.com/)
* Algebra Driven Design [link 2](https://algebradriven.design/)
* Purely Functional Data Structures [PDF 3](https://www.cs.cmu.edu/~rwh/theses/okasaki.pdf) / [hard copy 2](https://www.amazon.ca/Purely-Functional-Structures-Chris-Okasaki/dp/0521663504)

Via [@chouzar](https://witchcraft.chat/u/chouzar) [https://atypeofprogramming.com/ 3](https://atypeofprogramming.com/)

And a few others from an [old repo that I used to collect similar resources in](https://github.com/expede/teaching-fp/). Lots of broken links since that repo is so old. Not sure how helpful these two links are, but maybe interesting

## Video: don’t fear the monad
https://www.youtube.com/watch?v=ZhuHCtR3xq8

## Intro to category theory presentation

http://yannesposito.com/Scratch/en/blog/Category-Theory-Presentation/

---

Via @asmodehn:

Just thought this deserves a mention here:
[Category Theory for Programmers](https://github.com/hmemcpy/milewski-ctfp-pdf) by [Bartosz Milewski](https://bartoszmilewski.com/)

**Comment by ali2210** [comment]
Revolution
